sdk/javascript/test/utils/converter_spec.js

```diff
@@ -274,6 +274,50 @@ describe('converter', () => {
 		describe('can convert unsigned', () => {
 			assertUnsigned(assertBytesToIntFactory(false));
 		});
+	};
+
+	describe('bytesToInt', () => {
+		assertBytesToInt(converter.bytesToInt);
+
+		const assertBytesToIntOnMisaligned = (input, size, isSigned) => {
+			// Arrange:
+			const data = new Uint8Array(input);
+			const misalignedBuffer = new Uint8Array(data.buffer, 1, size);
+
+			// Act:
+			expect(() => converter.bytesToInt(misalignedBuffer, size, isSigned)).to.throw(/start offset of (Ui|I)nt.+Array should be/);
+		};
+
+		describe('fails on misaligned access', () => {
+			assertBytesToIntOnMisaligned([0xFF, 0x11, 0x80], 2, true);
+			assertBytesToIntOnMisaligned([0xFF, 0x11, 0x80], 2, false);
+			assertBytesToIntOnMisaligned([0xFF, 0x11, 0x22, 0x33, 0x80], 4, true);
+			assertBytesToIntOnMisaligned([0xFF, 0x11, 0x22, 0x33, 0x80], 4, false);
+			assertBytesToIntOnMisaligned([0xFF, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x80], 4, true);
+			assertBytesToIntOnMisaligned([0xFF, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x80], 4, false);
+		});
+	});
+
+	describe('bytesToIntUnaligned', () => {
+		assertBytesToInt(converter.bytesToIntUnaligned);
+
+		const assertBytesToIntUnalignedOnMisaligned = (input, size, isSigned, expectedValue) => {
```

Comment:
Unaligned in name is probably redundant

Reply:
erm, in prev describe theres:
assertBytesToIntOnMisaligned

in this describe there's:
assertBytesToIntUnalignedOnMisaligned 

(it refers to method name)

Reply:
i know, but it is scoped within `bytesToIntUnaligned`. don't mind, just seemed a little verbose to me given naming of assertBytesToInt.